sklearn/utils/metaestimators.py

```diff
@@ -87,10 +87,11 @@ class _IffHasAttrDescriptor(object):
     See https://docs.python.org/3/howto/descriptor.html for an explanation of
     descriptors.
     """
-    def __init__(self, fn, delegate_names, attribute_name):
+    def __init__(self, fn, delegate_names, attribute_name, backup_attribute):
```

Comment:
Here attribute_name should become a list, like delegate_names.

However I wonder if we should just use a property rather than making magic.